bgpd/bgp_fsm.c

```diff
@@ -588,6 +599,17 @@ bgp_stop_with_error (struct peer *peer)
   if (peer->v_start >= (60 * 2))
     peer->v_start = (60 * 2);
 
+  if (peer_dynamic_neighbor(peer))
+  {
+    if (BGP_DEBUG (events, EVENTS))
+    {
+      zlog_debug ("BGP stop with error: %s (dynamic neighbor) deleted", peer->host);
+    }
+
+    peer_delete (peer);
+    return -1;
+  }
+
```

Comment:
I think this logic can be merged into bgp_stop() as bgp_stop already has similar logic. We should enable code re-use.